### PR TITLE
Add updating of *.json and uploading of image files if exists to Simple Ingest

### PR DESCRIPTION
**Pull request by elv-wayne**
Add utilities/NFTIngest.js based on SimpleIngest.js
This fix facilitates generative video nft ingest where you are given the following set of source files:

nft001.mp4 // Input file
nft001.json  //Json file with nft attributes
nft001.jpg  //

Example source json:
```
{
  "attributes": [
    {
      "trait_type": "Team",
      "value": "Good"
    },
    {
      "trait_type": "Character",
      "value": "Prince"
    } ]
}
```

Example command:
`node utilities/NFTIngest.js --files /path_to_files/nft001.mp4 --libraryId ilibXXX--title "The title" --configUrl https://XXX.contentfabric.io/config --description "The description"`

IMAGE FILE:
The command will do the simple ingest and set object permission normally but also looks for
the corresponding image file nft001.jpg (can be any one of ["gif", "jpg", "jpeg", "png", "svg", "webp"]) and uploads it and creates a link to it under `public/asset_metadata/images/img/default`

JSON FILE:
the nft001.json file with if exists gets merges with new attributes: image, embed_url, name, display_name and description.

New nft001.json file would have:
```
{
  "attributes": [
    {
      "trait_type": "Team",
      "value": "Good"
    },
    {
      "trait_type": "Character",
      "value": "Prince"
    } ],
  "name": "test013",
  "display_name": "test013",
  "description": "test013 media description",
  "image": "https://host-66-220-3-88.contentfabric.io/q/hq__2MMx46i6JUK7iMUVDLtwRs6pZBaqCnUkDvEPdZPWRgqAu5tSqcQgX82ZBjWHyJnc48uNGJK1t9/meta/public/asset_metadata/images/img/default",
  "embed_url": "https://embed.v3.contentfabric.io/?p=&net=demo&vid=hq__2MMx46i6JUK7iMUVDLtwRs6pZBaqCnUkDvEPdZPWRgqAu5tSqcQgX82ZBjWHyJnc48uNGJK1t9&m=&ap=&lp="
}
```

Once all the json files are produced. There is a elv-live-js command to update an nft template object with generative nfts.

`./elv-live nft_build ilibXXX iq__XXX --nft_dir /path/to/json/files`

### CODE REVIEW

**Review by eponymous301 — CHANGES_REQUESTED** [review]
Wayne - could you copy simple ingest and make a new script (e.g. utilities/NFT_Ingest.js ?) with these changes? I'd like to keep the original script true to its purpose (simple media ingest, nothing extra)

File: utilities/SimpleIngest.js [hunk 1]
```diff
@@ -48,16 +57,27 @@ class SimpleIngest extends Utility {
...
 
-    const libMezPermission = R.path(["metadata", "abr", "mez_permission_level"], libInfo);
+    const libMezPermission = R.path(
+      ["metadata", "abr", "mez_permission_level"],
+      libInfo
+    );
 
-    const {drm, libraryId, title} = this.args;
+    const { drm, libraryId, title, configUrl, description } = this.args;
     const encrypt = true;
```
Comment (eponymous301) [inline utilities/SimpleIngest.js:81]: encrypt should be false for NFTs?

File: utilities/SimpleIngest.js [hunk 2]
```diff
@@ -266,20 +329,79 @@ class SimpleIngest extends Utility {
...
+        logger.log("Json file updated: ", jsonString);
+        //file written successfully
+      } catch (err) {
+        logger.warn("Could not read or write to json file", err);
+      }
+    }
   }
 
   header() {
     return "Create playable media object via simple ingest";
```
Comment (eponymous301) [inline utilities/SimpleIngest.js:462]: change to 'Ingest media for NFT'

Comment (elv-wayne) [reply utilities/SimpleIngest.js]: Done

File: utilities/SimpleIngest.js (hunk 1 above)
Comment (elv-wayne) [reply utilities/SimpleIngest.js:72]: Latest makes encrypt a command line param with default being "false"

**Comment by elv-wayne** [comment]
> 

Reverted SimpleIngest.js to original and created NFTIngest.js

**Review by eponymous301 — APPROVED** [review]
Thanks!

**Pull request merged**